filebeat/prospector/prospector.go

```diff
@@ -159,8 +159,8 @@ func (p *Prospector) Run(once bool) {
 // All state updates done by the prospector itself are synchronous to make sure not states are overwritten
 func (p *Prospector) updateState(event *input.Event) error {
 
-	// Add ttl if cleanOlder is enabled
-	if p.config.CleanInactive > 0 {
+	// Add ttl if cleanOlder is enabled and TTL is not already 0
+	if p.config.CleanInactive > 0 && event.State.TTL != 0 {
```

Comment:
The idea here is that TTL is reset every time the state is updated. As all new states are initialies with -1 this would mean these states would never get a TTL according to `clean_inactive`. My first idea was to have `state.Remove = true` as a flag for states that can be removed but it would blow up the change. But this would clearly show the difference between TTL and states that should be removed. TTL = 0 is like an alias for the Removed option and is only set when one of the clean options is reached.